File: oops/inheritance.py
#child inherit parent features
#we can access parent using parent class name in child class

class Parent:
    def phone(self):
        print("i have samsung j7")

class Child(Parent):
    pass

c=Child()
c.phone()

print(c)
print(id(c))

#different type of inheritance
#parent-->child
#super-->sub
#base-->derived

File: javascript/inheritance.js
class Parent{
    m1(){
        console.log("inside parent class");
    }
}

class Child extends Parent{
    m2(){
        console.log("inside child class");
    }
}

class SubChild extends Child{
    m3(){
        console.log("inside subchild class");
    }
}

var child=new SubChild()
child.m2();
child.m1();
child.m3()